### PR TITLE
Task 2: Holdings Table (Darell and Ricky)

### PR DESCRIPTION
**Pull request by darrll27**
Completed functional table and a function to add data to the table.

### CODE REVIEW

File: frontend/src/.DS_Store
Comment (JamesZhang878) [inline frontend/src/.DS_Store:1]: Can we also delete this please. Thank you

**Review by JamesZhang878 — CHANGES_REQUESTED** [review]
Thank you Darell! Very good work.

Here are a few thing that I noticed:
1. The Prices (Price Change + Market Price)are not being sorted correctly. 
- In the example data provided with the component, the price for APPL is labeled red, which I understand to mean a price change of -$120. However, when I try to sort the column, this row does not go to the bottom of the table.
- This issue affects both the Price Change and Market Price columns. It may also affect the Day Change column)
2. Home Page
- Currently, going to the main page at localhost:3000 still takes us to the default stater page for Next.js. Could you change this to the holdings page?
3. Long File
- The holdings-table file is currently very long. Can we restructure it by moving everything that is not the HoldingsTable() function into different files and import them in? For example, let's put the placeholder dummy data in a separate file and have fetch functions temporarily use data imported from those files. These fetch functions will eventually call the API endpoints written in #12 
4. Miscellaneous code
- I noticed that there are lots of commented out code, un-used imports along with .DS_Store files in this PR. Can we delete those please. 

Other than that, the table looks great. Thanks again Darell!

**Comment by IBN-5100** [comment]
Thanks for the opportunity to work on the code. Looks great. I've completed the assign tasks to this degree:

1. Price Sorting Bug
Fully resolved, this was an issue with mathematically impossible data. I also cleaned up and used parentheses to represent negative values for price change because that seems to be the agreed standard notation from day change and gain/loss. Now the gain/loss also matches up with quantity * change as it should.

2. Home Page
Resolved. Removed the boilerplate default nextjs home page. Replaced with barebones SFIC home page that directs the user to the Holdings page. Also includes links to other pages for future.

3. Long File
Not Resolved. Cleaned redundant comments and length.
TODO: New data pipeline. Take out cluttered functions and import in as needed.

4. Miscellaneous Code
Resolved. I have removed all .DS_Store files and most redundant or superfluous code comments.

**Review by JamesZhang878 — CHANGES_REQUESTED** [review]
Thanks for all of the changes @IBN-5100! The new landing page looks great :)

I did notice that there is a small issue with paging. Specifically, the rows aren't being displayed correctly. I discovered this issue after adding more rows to test the pagination. 

Basically, what's happening is that when I switch from the first page to the second page and then back to the first page again, the contents loaded in from the second page aren't being cleared out. I suspect it has something to do with the way that the "selected" state declared on line 245 of holdings-table.tsx is being used.

Could you take another look into this? Thanks!

**Comment by IBN-5100** [comment]
1.  ✅Pagination/sorting causing duplicating entries has been fixed! As discussed in meeting, it was an issue with duplicate ids, and the sample data has been corrected. 

2. 🚧Additionally, to prevent this issue from happening again, some basic error handling for **_duplicate ids_** and **_valid data_** has been added into the temporary constants/data but can be moved into a component in a later build.

3. 🚧Also added a basic popup component that can be used to deliver real error messages (non lethal) in later builds.

Error handling currently only writes to console.

**Pull request merged**